types/tests/relationships-test.ts

```diff
@@ -34,6 +27,9 @@ const people = schema.all("person");
 people.length; // $ExpectType number
 people.modelName; // $ExpectType string
 people.models.map((model) => {
+  model.nothing; // $ExpectType unknown
```

Comment:
(Thinking out loud.) This makes me wonder about the tradeoff—the types mean we end up with `unknown` here; we could also restructure so it resolves as `never`. `unknown` is certainly more flexible, and for a codebase in transition to TS likely more *accurate*, as some types won't be provided yet. Definitely a thing for a later point, but it might be interesting to supply an alternative set of imports and give users the ability to use imports to specify which variant they want – something like `import 'miragejs/strict';` with the types resolving to `never`. 🤔 Definitely, definitely does not need to be a change here, though.

Reply:
If it resolves as `never`, isn't that effectively the same as resolving to `any` since it's in a covariant position? Unless we did a mapping pass that eliminated keys whose values were `never`, though the failure mode for that seems more confusing to me from the user perspective ("why does it say this key doesn't exist on my model? I defined it right there!")

We could introduce some kind of `InvalidModelName` sigil or something to resolve it to instead of `unknown`, if that seems like it would be more helpful in nudging people to understand what they did wrong 🤔

Reply:
Yeah, you’re right. And anyway you’d want the error to happen at the definition site, not the use site.